demos/noise_suppression_demo/cpp/main.cpp

```diff
@@ -2,67 +2,45 @@
-* \example noise_suppression_demo/main.cpp
-*/
-#include <climits>
-#include <vector>
-#include <string>
-#include <chrono>
-#include <iomanip>
-#include <iostream>
-#include <fstream>
-
-#include "openvino/openvino.hpp"
-
-#include "gflags/gflags.h"
-#include "utils/common.hpp"
-#include "utils/slog.hpp"
-
-typedef std::chrono::steady_clock Time;
-
-static const char help_message[] = "Print a usage message.";
-static const char inp_wav_message[] = "Required. Path to a input WAV file.";
-static const char out_wav_message[] = "Optional. Path to a output WAV file.";
-static const char model_message[] = "Required. Path to an .xml file with a trained model.";
-static const char target_device_message[] = "Optional. Specify the target device to infer on (the list of available "
-                                            "devices is shown below). Default value is CPU. "
-                                            "The demo will look for a suitable plugin for device specified.";
-DEFINE_bool(h, false, help_message);
-DEFINE_string(i, "", inp_wav_message);
-DEFINE_string(o, "noise_suppression_demo_out.wav", out_wav_message);
-DEFINE_string(m, "", model_message);
-DEFINE_string(d, "CPU", target_device_message);
-
-static void showUsage() {
-    std::cout << std::endl;
-    std::cout << "noise_suppression_demo [OPTION]" << std::endl;
-    std::cout << "Options:" << std::endl;
-    std::cout << std::endl;
-    std::cout << "    -h           " << help_message << std::endl;
-    std::cout << "    -i INPUT     " << inp_wav_message << std::endl;
-    std::cout << "    -o OUTPUT    " << out_wav_message << std::endl;
-    std::cout << "    -m MODEL     " << model_message << std::endl;
-    std::cout << "    -d DEVICE    " << target_device_message << std::endl;
-}
-
-bool ParseAndCheckCommandLine(int argc, char* argv[]) {
-    gflags::ParseCommandLineNonHelpFlags(&argc, &argv, true);
-    if (FLAGS_h) {
-        showUsage();
-        return false;
+#include <openvino/openvino.hpp>
+#include <gflags/gflags.h>
+#include <utils/common.hpp>
+#include <utils/slog.hpp>
+
+namespace {
+constexpr char h_msg[] = "show the help message and exit";
+DEFINE_bool(h, false, h_msg);
+
+constexpr char m_msg[] = "path to an .xml file with a trained model";
+DEFINE_string(m, "", m_msg);
+
+constexpr char i_msg[] = "path to an input 16kHz WAV file";
+DEFINE_string(i, "", i_msg);
+
+constexpr char d_msg[] = "specify a device to infer on (the list of available devices is shown below). Default is CPU";
+DEFINE_string(d, "CPU", d_msg);
+
+constexpr char o_msg[] = "path to an output WAV file. Default is noise_suppression_demo_out.wav";
+DEFINE_string(o, "noise_suppression_demo_out.wav", o_msg);
+
+void parse(int argc, char *argv[]) {
+    gflags::ParseCommandLineFlags(&argc, &argv, false);
+    if (FLAGS_h || 1 == argc) {
+        std::cout <<   "\t[-h]               " << h_msg
+                  << "\n\t[--help]           print help on all arguments"
```

Comment:
`--help` is a bit redundant in C++, isn't it?

Reply:
gflags generate --help which behaves differently compared to our -h

Reply:
Yeah, I see that not only description of flags from the demo is generated, but also some default flags from gflags\src. If it might be useful, then okay